arch/x86/core/x86_mmu.c

```diff
@@ -25,6 +25,8 @@ MMU_BOOT_REGION((u32_t)&_image_rom_start, (u32_t)&_image_rom_size,
 MMU_BOOT_REGION((u32_t)&__app_ram_start, (u32_t)&__app_ram_size,
 		MMU_ENTRY_WRITE | MMU_ENTRY_USER | MMU_ENTRY_EXECUTE_DISABLE);
 #endif
+MMU_BOOT_REGION((u32_t)&_app_smem_start, (u32_t)&_app_smem_size,
+		MMU_ENTRY_WRITE | MMU_ENTRY_USER | MMU_ENTRY_EXECUTE_DISABLE);
```

Comment:
Does this need to be wrapped under CONFIG_APP_SHARED_MEM?

Reply:
Agreed, this has to be inside a ifdef block.